lib/facter/resolvers/partitions.rb

```diff
@@ -122,8 +122,15 @@ def execute_and_extract_blkid_info
         def populate_from_lsblk(partition_name, blkid_and_lsblk)
           return {} unless available?('lsblk', blkid_and_lsblk)
 
-          lsblk_version_raw = Facter::Core::Execution.execute('lsblk --version 2>&1', logger: log)
-          lsblk_version = lsblk_version_raw.match(/ \d\.\d+/)[0].to_f
+          # Catch exceptions and return for old versions of lsblk (< 2.22) that do not support the --version flag
+          # We only really need this for one of the oldest OSes that Puppet supports, SLES 11. We can probably remove this error handling
+          # when we drop SLES 11 support in Puppet.
+          begin
+            lsblk_version_raw = Facter::Core::Execution.execute('lsblk --version 2>&1', logger: log)
+            lsblk_version = lsblk_version_raw.match(/ \d\.\d+/)[0].to_f
+          rescue NoMethodError
```

Comment:
Does the `execution` method return an exit code so you can know if it succeeded or not? I'm thinking the NoMethodError is raised when calling `[0]` on nil. Maybe check for a match explicitly?

Reply:
While `lsblk --version` returns 1 when run in Bash, it doesn't look like `Facter::Core::Execution.execute` returns any exit code:

https://github.com/puppetlabs/facter/blob/b6cf607a746e02e040ce5ca2dc85cc14c0982803/lib/facter/custom_facts/core/execution/base.rb#L110

Reply:
Would it make sense to handle the nil case explicitly?

```ruby
lsblk_version_raw = ...
lsblk_version = lsblk_version_raw.match(/ \d\.\d+/)&.first&.to_f || 0
return {} if lsblk_version < 2.23
```